model/scratch/dffml_model_scratch/model/slr.py

```diff
@@ -74,11 +75,12 @@ def get_regression_line(self):
         prediction = self.regression_line[0]*x+self.regression_line[1]
         return prediction
 
-    async def train(self, sources: Sources, features: Features):
+    async def train(self, sources: Sources, features: Features, _classifications):
         '''
         Train using repos as the data to learn from.
         '''
         feature = await self.applicable_features(features)
+        self.key = hashlib.sha384((''.join(sorted(feature))).encode()).hexdigest()
```

Comment:
Try following the pattern that I set out with `get_regression_line` and `set_regression_line` please, multiple of these methods could be called at the same time and that would create problems if you use `self.key` like this

Reply:
should I use a helper method then? or the way you implemented

Reply:
Any suggestions on how would I pass features on function calls? __init__ and best_fit_line method. Latter is understandable but I think there might be a better...

Reply:
Try this:

https://github.com/intel/dffml/pull/96#discussion_r293017252

Reply:
I did, but function calls are in __init__ where I don't have access to features